pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,144 @@ $title: Use custom JavaScript in AMP pages
+[Web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) present a way to run JavaScript more safely. Normally all JavaScript [runs in a single thread](https://www.youtube.com/watch?v=cCOL7MC4Pl0), but each worker runs in a thread of its own. This is possible because they lack access to the DOM or the `window` object, and each worker runs in its own global scope. Thus they can't interfere with each other's work or with mutations caused by code in the main thread. They can only communicate with the main thread and with one another via [messages containing objects](https://developer.mozilla.org/en-US/docs/Web/API/DedicatedWorkerGlobalScope/postMessage). Workers offer a path to a multithreaded Web, a way to encapsulate JavaScript in a sandbox where it can't block the UI.
 
-## Custom scripts caching
+Workers don't come with access to the DOM. To fill this gap, the AMP team created an open-source library called [WorkerDOM](https://github.com/ampproject/worker-dom/). WorkerDOM copies the DOM to a virtual DOM and makes the copy available to a worker. WorkerDOM also recreates a subset of the standard DOM API. When the worker makes changes to the virtual DOM, WorkerDOM recreates those changes in the real DOM. This lets the worker manipulate the DOM and make changes on the page using standard techniques. The DOM synchronization only goes in one direction. If the main thread modifies the DOM, nothing will let the worker know.
 
-The [AMP cache](../../../documentation/guides-and-tutorials/learn/amp-caches-and-cors/how_amp_pages_are_cached.md) serves custom JavaScript files inserted with `<amp-script>` the same way it serves AMP component scripts, which ensures that any custom JavaScript will not slow the speed of an AMP document.
+`amp-script` is essentially a wrapper around WorkerDOM that makes WorkerDOM usable in AMP. WorkerDOM provides the core of `amp-script`'s functionality.
 
-The AMP cache proxies the JavaScript files and then delivers them. Users can expect the same performance experience from a page using `<amp-script>` as a page that doesn't include it.
 
-# Using `<amp-script>`
+## What `amp-script` can do
 
-To guarantee AMP pages will load consistently fast and with smooth UIs, limitations exist on `<amp-script>`.
+The JavaScript language is the same in a worker as it is elsewhere in the browser. Thus, in `amp-script`, you can use JavaScript in the way that you're used to. WorkerDOM also recreates many commonly used DOM APIs and makes them available for your use. It supports common Web APIs like `Fetch` and `Canvas`, and it gives you to selected global objects like `navigator` and `localStorage`. You can assign handlers for browser events in the usual way.
```

Comment:
I'm not really sure what "use JavaScript in the way that you're used to" means? And I'm not sure it's true - developers need to use workarounds to create certain functionalities?

Reply:
JavaScript is the same language as always, but the APIs the browser provides are different. You can I had trouble explaining this.